### PR TITLE
Added example for one-hot encoded datasets

### PR DESCRIPTION
**Pull request by sdraper-CS**
See https://github.com/CognitiveScale/certifai/issues/2039

Note that this is a PR to `pre-release` because this example requires the latest toolkit.

### CODE REVIEW

File: notebooks/one_hot_encoded/one_hot_encoded.ipynb [hunk 1]
```diff
@@ -0,0 +1,939 @@
...
+    "        \n",
+    "# Save to an encoded CSV \n",
+    "df_encoded.to_csv(all_data_one_hot_file, index=False)\n",
+    "\n",
+    "# This is what the data looks like after encoding\n",
+    "display(df_encoded)\n",
+    "\n",
+    "# We will also want to know how the categorical encoding has mapped categorical values to columns, so\n",
+    "# save this as a dictionary for later use (if the data were given to us pre-encoded then we'd need\n",
+    "# to construct this mapping manually based on manual analysis of the data or a schem defintion also given\n",
```
Comment (npasricha-cs) [inline notebooks/one_hot_encoded/one_hot_encoded.ipynb:671]: schem -> schema

Appears an added "." as well

File: notebooks/one_hot_encoded/one_hot_encoded.ipynb [hunk 2]
```diff
@@ -0,0 +1,939 @@
...
+    "#. to us with the dataset)\n",
```
Comment (npasricha-cs) [inline notebooks/one_hot_encoded/one_hot_encoded.ipynb:672]: Not sure if "." should be here

File: notebooks/one_hot_encoded/one_hot_encoded.ipynb (hunk 1 above)
Comment (sdraper-CS) [reply notebooks/one_hot_encoded/one_hot_encoded.ipynb:671]: Fixed

File: notebooks/one_hot_encoded/one_hot_encoded.ipynb (hunk 2 above)
Comment (sdraper-CS) [reply notebooks/one_hot_encoded/one_hot_encoded.ipynb:672]: Removed

**Comment by cdraper-CS** [comment]
It would be useful to point out in the intro text that:
1. It's important to define the one-hot encodings and not just pass the data to Certifai, because otherwise Certifai will send the model invalid prediction requests
2. A benefit of this is that the results (e.g. explanations and I assume fairness burdens) are presented in terms of meaningful features/values. 

Did you test it works when fairness feature is one-hot encoded? Do  we need an example of that, either here or in doc?

Note this will need a PR on scan definition reference docs:
https://github.com/CognitiveScale/cortex-certifai/blob/master/docs/docs/reference/scan-definition.md

**Pull request merged**